src/main/java/io/cryostat/rules/Rules.java

```diff
@@ -147,9 +156,13 @@ public RestResponse<V2Response> delete(@RestPath String name, @RestQuery boolean
         return RestResponse.ok(V2Response.json(Response.Status.OK, null));
     }
 
-    static class RuleExistsException extends ClientErrorException {
+    public class RuleExistsException extends ClientErrorException {
```

Comment:
I think this should be public static - no need for the exception to carry an implicit reference to the `Rules` instance it was instantiated within.